nestkernel/connection_label.h

```diff
@@ -0,0 +1,113 @@
+/*
+ *  connection_label.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef CONNECTION_LABEL_H
+#define CONNECTION_LABEL_H
+
+#include "nest.h"
+#include "nest_names.h"
+#include "dictdatum.h"
+#include "dictutils.h"
+
+namespace nest
+{
+class ConnectorModel;
+
+/**
+ * Connections are unlabeled by default. Unlabeled connections cannot be specified
+ * as a search criterion in the `GetConnections` function.
+ * @see ConnectionLabel
+ */
+const long_t UNLABELED_CONNECTION = -1;
```

Comment:
I agree!

> On 08 Dec 2015, at 13:50, Tammo Ippen <notifications@github.com> wrote:
> 
> In nestkernel/connection_label.h:
> 
> > +
> > +#include "nest.h"
> > +#include "nest_names.h"
> > +#include "dictdatum.h"
> > +#include "dictutils.h"
> > +
> > +namespace nest
> > +{
> > +class ConnectorModel;
> > +
> > +/**
> > + * Connections are unlabeled by default. Unlabeled connections cannot be specified
> > + * as a search criterion in the `GetConnections` function.
> > + * @see ConnectionLabel
> > + */
> > +const long_t UNLABELED_CONNECTION = -1;
> 
> This would require a template argument, whenever you want to access this field, like:
> 
> ConnectionLabel< StaticConnection< TargetIdentifierPtrRport > >::UNLABELED_CONNECTION
> 
> I guess we should not go for this...
> 
> —
> Reply to this email directly or view it on GitHub.
> 


--

Dr. Hans Ekkehard Plesser, Associate Professor
Section Head

Dept. of Mathematical Sciences and Technology
Norwegian University of Life Sciences
PO Box 5003, 1432 Aas, Norway

Phone +47 6723 1560
Email hans.ekkehard.plesser@nmbu.no
Home  http://arken.umb.no/~plesser